test/end-to-end/auth.js

```diff
@@ -199,5 +199,13 @@ describe("Login endpoint test", () => {
 
             expect(res.status).toBe(HTTPStatus.UNAUTHORIZED);
         });
+
+        test("should return OK since the logout is idempotent", async () => {
+            const res = await test_agent
+                .delete("/auth/login")
+                .send();
+
+            expect(res.status).toBe(HTTPStatus.OK);
```

Comment:
This would pass even if it were not idempotent and for some reason the agent was logged in due to some other test.

I'd wrap this test in a describe block called "clean session" or something that would define a new test_agent just for the test (thus isolating it from the agent used in other tests, and being logged out by default). You can even rename the agent variable to logged_out_agent to make it clear.

Reply:
I think I fixed it, but I'm just now using this framework, so please review it again.

Reply:
Just created a simple scenario to confirm this would work:

```js
describe("Using logged out user", () => {
        const logged_out_agent = agent();
        const logged_in_agent = agent();

        beforeAll(async () => {
            const user = {
                email: "user@email.com",
                password: "password123",
            };

            const res = await request()
                .post("/auth/register")
                .send(withGodToken(user));

            console.log("REG RESULT", res.body);
        });

        test("should return OK since the logout is idempotent", async () => {
            await logged_in_agent
                .post("/auth/login")
                .send({
                    email: "user@email.com",
                    password: "password123",
                });

            const res = await logged_out_agent
                .get("/auth/me")
                .send();

            console.log(res.body);
            expect(res.status).toBe(HTTPStatus.OK);
        });
    });
```

And it failed, as expected, confirming that a new user (agent) is indeed used, so your implementation should work